deploy-asm.sh

```diff
@@ -33,40 +33,30 @@ echo "No CLUSTERNAME variable set"
 exit
 fi
 
-gcloud services enable cloudtrace.googleapis.com
-
-gcloud container clusters get-credentials $CLUSTERNAME --project=$PROJECT --zone=$LOCATION
-kubectl config set-context $CLUSTERNAME
-
-echo "Installing ASM..."
-curl -s https://storage.googleapis.com/csm-artifacts/asm/asmcli_1.13 > asmcli
-chmod +x asmcli
+echo "Register GKE cluster to the fleet..."
+gcloud container fleet memberships register $CLUSTERNAME \
+    --gke-cluster $LOCATION/$CLUSTERNAME \
+    --enable-workload-identity
+    
+echo "Enable Managed ASM on the project..."
+gcloud container fleet mesh enable
 
 sleep 5s
 
-./asmcli install \
-  --project_id $PROJECT \
-  --cluster_name $CLUSTERNAME \
-  --cluster_location $LOCATION \
-  --fleet_id $PROJECT \
-  --output_dir asmoutput \
-  --enable_all \
-  --ca mesh_ca
+echo "Enable Managed ASM on the GKE cluster..."
+gcloud container fleet mesh update \
+    --management automatic \
+    --memberships $CLUSTERNAME
 
-sleep 10s
+sleep 10s # we need to wait more here...
 
-export REVISION=$(kubectl get deploy -n istio-system -l app=istiod -o jsonpath={.items[*].metadata.labels.'istio\.io\/rev'}'{"\n"}')
+gcloud container clusters get-credentials $CLUSTERNAME --project=$PROJECT --zone=$LOCATION
+kubectl config set-context $CLUSTERNAME
 
 echo "Deploying Online Boutique sample application..."
-kubectl apply -f asmoutput/samples/online-boutique/kubernetes-manifests/namespaces
-
-for ns in ad cart checkout currency email frontend loadgenerator payment product-catalog recommendation shipping; do
-  kubectl label namespace $ns istio.io/rev=$REVISION --overwrite
-done;
-
-kubectl apply -f asmoutput/samples/online-boutique/kubernetes-manifests/deployments
-kubectl apply -f asmoutput/samples/online-boutique/kubernetes-manifests/services
-kubectl apply -f asmoutput/samples/online-boutique/istio-manifests/allow-egress-googleapis.yaml
+kubectl create namespace onlineboutique
+kubectl label namespace onlineboutique istio-injection=enabled
+kubectl apply -f https://github.com/GoogleCloudPlatform/microservices-demo/raw/main/release/kubernetes-manifests.yaml -n onlineboutique
```

Comment:
Rather than using the main, can we use a release tag instead?

- So should we stick to a particular release or can we just automatically pick the latest release using the GitHub API (https://api.github.com/repos/GoogleCloudPlatform/microservices-demo/releases/latest) and get the latest release?

Reply:
Yeah, good point, I can update that part to get the `latest` release version and get the associated manifests, makes sense. FYI: anyway that release folder/manifest doesn't change between 2 releases, but that's safer the way you are referring, will do that soon.

Reply:
Done.